src/SignalR/clients/cpp/include/signalrclient/connection.h

```diff
@@ -29,16 +28,16 @@ namespace signalr
 
         connection& operator=(const connection&) = delete;
 
-        SIGNALRCLIENT_API pplx::task<void> __cdecl start();
+        SIGNALRCLIENT_API void __cdecl start(std::function<void(std::exception_ptr)> callback) noexcept;
 
-        SIGNALRCLIENT_API pplx::task<void> __cdecl send(const std::string& data);
+        SIGNALRCLIENT_API void __cdecl send(const std::string& data, std::function<void(std::exception_ptr)> callback) noexcept;
 
         SIGNALRCLIENT_API void __cdecl set_message_received(const message_received_handler& message_received_callback);
         SIGNALRCLIENT_API void __cdecl set_disconnected(const std::function<void __cdecl()>& disconnected_callback);
```

Comment:
Would it make sense to have set_disconnected's completion callback take an exception as well?

Reply:
Probably, I'll track that in our public API review issue https://github.com/aspnet/AspNetCore/issues/8717